maven/lib/dependabot/maven/native_helpers.rb

```diff
@@ -0,0 +1,42 @@
+# typed: strong
+# frozen_string_literal: true
+
+require "shellwords"
+require "sorbet-runtime"
+
+module Dependabot
+  module Maven
+    module NativeHelpers
+      extend T::Sig
+
+      sig do
+        params(file_name: String).void
+      end
+      def self.run_mvn_dependency_tree_plugin(file_name)
+        proxy_url = URI.parse(ENV.fetch("HTTPS_PROXY"))
+        stdout, _, status = Open3.capture3(
+          { "PROXY_HOST" => proxy_url.host },
+          "mvn",
+          "dependency:tree",
+          "-DoutputFile=#{file_name}",
+          "-DoutputType=json",
+          "-e"
+        )
+        Dependabot.logger.info("mvn dependency:tree output: STDOUT:#{stdout}")
+        handle_tool_error(stdout) unless status.success?
+      end
+
+      sig { params(output: String).void }
+      def self.handle_tool_error(output)
+        if (match = output.match(
+          %r{Could not transfer artifact (?<artifact>[^ ]+) from/to (?<repository_name>[^ ]+)
+           \((?<repository_url>[^ ]+)\): status code: (?<status_code>[0-9]+)}
+        )) && (match[:status_code] == ("403") || match[:status_code] == ("401"))
+          raise Dependabot::PrivateSourceAuthenticationFailure, match[:repository_url]
+        end
+
+        raise DependabotError, "mvn CLI failed with an unhandled error"
```

Comment:
I feel like we ought to print/include the stdout and stderr in this case.

Reply:
I already print maven cli stdout earlier on line 25 regardless of success or not, so it's easier to debug and understand what's going on. From my testing maven outputs everything into stdout and stderr is always blank.